diff --git a/UI/AppKit/Interface/TabController.mm b/UI/AppKit/Interface/TabController.mm
index 04aa472b33..0dd8eaba1c 100644
--- a/UI/AppKit/Interface/TabController.mm
+++ b/UI/AppKit/Interface/TabController.mm
@@ -1495,7 +1495,7 @@ private:
 - (BOOL)windowShouldClose:(NSWindow*)sender
 {
     auto* delegate = (ApplicationDelegate*)[NSApp delegate];
-    auto confirm_canceling_downloads = [&]() {
+    auto confirm_canceling_downloads = [&]() -> bool {
         if ([delegate tabCount] > 1)
             return true;
         return [(Application*)NSApp confirmCancelActiveDownloads];
